src/utils.ts

```diff
@@ -0,0 +1,192 @@
+import {
+  AccountId,
+  AssetValue,
+  AztecSdk,
+  EthAddress,
+  EthereumProvider,
+  GrumpkinAddress,
+  SchnorrSigner,
+  TxSettlementTime,
+  TxId,
+  WalletProvider,
+  Web3Signer,
+  RegisterController,
+  DepositController,
+  DefiController,
+  BridgeCallData,
+  Signer,
+  AztecSdkUser,
+} from "@aztec/sdk";
+import { Web3Provider } from "@ethersproject/providers";
+
+const privateKeyMessage = Buffer.from(
+  `Sign this message to generate your Aztec Privacy Key. This key lets the application decrypt your balance on Aztec.\n\nIMPORTANT: Only sign this message if you trust the application.`
+);
+
+const spendingKeyMessage = Buffer.from(
+  `Sign this message to generate your Aztec Spending Key. This key lets the application spend your funds on Aztec.\n\nIMPORTANT: Only sign this message if you trust the application.`
+);
+
+export async function createSpendingKey(
+  provider: EthereumProvider,
+  sdk: AztecSdk
+) {
+  const privateKey = await createSigningKey(provider, spendingKeyMessage);
+  const publicKey = await sdk.derivePublicKey(privateKey);
+  return { privateKey, publicKey };
+}
+
+export async function createPrivacyKey(
+  provider: EthereumProvider,
+  sdk: AztecSdk
+) {
+  const privateKey = await createSigningKey(provider, privateKeyMessage);
+  const publicKey = await sdk.derivePublicKey(privateKey);
+  return { privateKey, publicKey };
+}
+
+export async function createArbitraryDeterministicKey(
+  provider: EthereumProvider,
+  sdk: AztecSdk,
+  message: string
+) {
+  const messageToSign = Buffer.from(message);
+  const privateKey = await createSigningKey(provider, messageToSign);
+  const publicKey = await sdk.derivePublicKey(privateKey);
+  return { privateKey, publicKey };
+}
+
+const createSigningKey = async (
+  provider: EthereumProvider,
+  message: Buffer
+) => {
+  const signer = new Web3Signer(provider);
+  const ethAddress = (await provider.request({ method: "eth_accounts" }))[0];
+  const signedMessage = await signer.signMessage(message, ethAddress);
+  return signedMessage.slice(0, 32);
+};
```

Comment:
Yes. Quite some legacy code is lying around pending for my cleanup 😂
Simply merging early in case if you'd like to start playing with it. We can work on my forked repo instead as well if that's preferred.

Reply:
Sorry, just saw this. No worries, I'm happy to merge. I'll do that now.